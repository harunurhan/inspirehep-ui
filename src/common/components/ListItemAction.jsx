import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import IconText from './IconText';


class ListItemAction extends Component {
  render() {
    const {
      iconType, text, href, onClick, target,
    } = this.props;

    return (
      <Button
        style={{ border: 'none', background: 'transparent' }}
        href={href}
        target={target}
        onClick={onClick}
      >
        <IconText text={text} type={iconType} />
      </Button>
    );
  }
}

ListItemAction.propTypes = {
  iconType: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  href: PropTypes.string,
  target: PropTypes.string,
  onClick: PropTypes.func,
};

ListItemAction.defaultProps = {
  onClick: undefined,
  href: undefined,
  target: undefined,
};

export default ListItemAction;
