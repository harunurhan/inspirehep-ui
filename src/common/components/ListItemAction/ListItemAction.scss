.__ListItemAction__ {
  border: none;
  background: transparent !important;

  .no-transition {
    transition: unset;
  }
}