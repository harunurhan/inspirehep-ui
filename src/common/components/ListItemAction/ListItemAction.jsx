import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button } from 'antd';

import IconText from '../IconText';
import './ListItemAction.scss';

class ListItemAction extends Component {
  wrapWithRouterLinkIfToPropSet(component) {
    const { link } = this.props;
    if (link.to) {
      return (
        <Link className="no-transition" to={link.to}>
          {component}
        </Link>
      );
    }
    return component;
  }

  render() {
    const {
      iconType, text, link, onClick,
    } = this.props;

    return (
      <Button
        className="__ListItemAction__"
        href={link.href}
        target={link.target}
        onClick={onClick}
      >
        {this.wrapWithRouterLinkIfToPropSet(<IconText text={text} type={iconType} />)}
      </Button>
    );
  }
}

ListItemAction.propTypes = {
  iconType: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  link: PropTypes.oneOfType([
    PropTypes.shape({ to: PropTypes.string.isRequired }),
    PropTypes.shape({ href: PropTypes.string.isRequired, target: PropTypes.string }),
  ]),
  onClick: PropTypes.func,
};

ListItemAction.defaultProps = {
  onClick: undefined,
  link: {},
};

export default ListItemAction;
