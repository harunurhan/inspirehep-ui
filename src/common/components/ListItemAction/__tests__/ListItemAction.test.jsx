import React from 'react';
import { shallow } from 'enzyme';

import ListItemAction from '../ListItemAction';

describe('ListItemAction', () => {
  it('renders with link (href and target)', () => {
    const wrapper = shallow((
      <ListItemAction
        iconType="info"
        text="Test"
        link={{
          href: '//example.com',
          target: '_blank',
        }}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });


  it('renders with link (to) by using Router.Link', () => {
    const wrapper = shallow((
      <ListItemAction
        iconType="info"
        text="Test"
        link={{
          to: '/relative/link',
        }}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with onClick', () => {
    const wrapper = shallow((
      <ListItemAction
        iconType="info"
        text="Test"
        onClick={jest.fn()}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
