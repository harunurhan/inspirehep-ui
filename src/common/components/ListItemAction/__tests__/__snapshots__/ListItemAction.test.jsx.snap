// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemAction renders with link (href and target) 1`] = `
<Button
  className="__ListItemAction__"
  ghost={false}
  href="//example.com"
  loading={false}
  prefixCls="ant-btn"
  target="_blank"
>
  <IconText
    text="Test"
    type="info"
  />
</Button>
`;

exports[`ListItemAction renders with link (to) by using Router.Link 1`] = `
<Button
  className="__ListItemAction__"
  ghost={false}
  loading={false}
  prefixCls="ant-btn"
>
  <Link
    className="no-transition"
    replace={false}
    to="/relative/link"
  >
    <IconText
      text="Test"
      type="info"
    />
  </Link>
</Button>
`;

exports[`ListItemAction renders with onClick 1`] = `
<Button
  className="__ListItemAction__"
  ghost={false}
  loading={false}
  onClick={[Function]}
  prefixCls="ant-btn"
>
  <IconText
    text="Test"
    type="info"
  />
</Button>
`;
